### PR TITLE
`General`: Do not show certain buttons for tutors in lecture overview and course management

### LINKED ISSUE #7422: `Lecture`: lecture unit creation page visible even if no permission
### Describe the bug

with tutor rights in a course I am able to navigate to the "lecture unit" creation page even if I'm not allowed to use it

### To Reproduce

Being tutor in a course
1. go to Lectures
2. open a lecture
3. click on the "Manage" button in the upper right
4. select the "Units" button at the bottom of the view
5. click on one of the unit types next to "Add a new Lecture Unit:"
6. now you'll see the unit creation form and an error messages saying "You are not authorized to access this page."

### Expected behavior

you should not be able to open the "add lecture unit" form since you, as a tutor, have no permission to use it anyways

### Screenshots

https://cdn.discordapp.com/attachments/1163529763361665084/1164481135817015386/image.png?ex=65435e91&is=6530e991&hm=5f7dfde5f052dbe1797d4468eafd04d6895a83958351b90cf7ba869b817b252f&

### Which version of Artemis are you seeing the problem on?

6.6.1

### What browsers are you seeing the problem on?

Firefox

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Fixes #7422 and also some other occurrences I noticed when fixing this issue.

### Description

Fixes authorities and hides the following buttons from tutors:
- Manage button in the lecture view inside a course (the one from #7422)
- "Edit"/"Attachment"/"Units" buttons in the manage page of a lecture
- Scores button in the course management tab bar
- In the general course management the "Tutorials" button in the course card, if there is no tutorial configuration yet

Also shows the tutorials button in the course management tab bar of a specific course so tutors have access to their tutor groups and can sign up students. This is only possible after an instructor has configured the tutorial groups.

### Steps for Testing

Prerequisites:
- 1 Tutor
- 1 Editor/Instruktor

1. Check that the buttons are successfully hidden from tutors in a course without a tutorial group configuration
2. Check that instructors see all buttons
3. Enable tutorial groups (the tutorials tab guides the instructor through this) or find a course with tutorial groups enabled
4. Check that tutors can now access tutorials in the course management
(Also see attached screenshots and links to examples on TS1)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

The hidden Buttons:

Manage button, e.g. here: https://artemis-test1.artemis.in.tum.de/courses/118/lectures/215
Bildschirmfoto 2023-10-19 um 17 04 08

"Edit"/"Attachment"/"Units" buttons, e.g. here: https://artemis-test1.artemis.in.tum.de/course-management/118/lectures/215
Bildschirmfoto 2023-10-19 um 17 04 19

Scores button and tutorials button not visible to tutors, e.g. here: https://artemis-test1.artemis.in.tum.de/course-management/118
Bildschirmfoto 2023-10-21 um 11 33 25

Tutorials button visible in tab bar, e.g. here: https://artemis-test1.artemis.in.tum.de/course-management/118
Bildschirmfoto 2023-10-21 um 11 34 59

Tutorials button visible in course management card, e.g. here: https://artemis-test1.artemis.in.tum.de/course-management under the test course section
Bildschirmfoto 2023-10-21 um 11 35 37

### CODE REVIEW

**Review by konrad2002 — APPROVED** [review]
LGTM
Tested on Legacy TS2

**Review by Strohgelaender — APPROVED** [review]
code lgtm, thanks for adding a small additional client test

**Review by MaximilianAnzinger — APPROVED** [review]
Code lgtm

**Review by dearjasmina — DISMISSED** [review]
tested on ts1, works as described for both tutor and instructor roles 👍🏻

**Comment by JanaNF** [comment]
Manual tested on ts1. Worked as expected

**Review by JanaNF — APPROVED** [review]
Tested on ts1. All good

**Review by konrad2002 — CHANGES_REQUESTED** [review]
tested it on legacy ts11

I'm not sure if it is supposed to be like this, but as a tutor I wasn't able to open the tutorial groups page..

As instructor this worked

### Tutor:
grafik
When clicking, nothing happens (+ no error in console)

### Instructor:
grafik
After clicking on tutorials

File: src/main/webapp/app/course/manage/overview/course-management-card.component.html [hunk 1]
```diff
@@ -254,7 +254,7 @@ <h4 class="text-center no-exercises mt-3 fw-medium">{{ 'artemisApp.course.noExer
             <span class="d-none d-xl-inline">{{ 'artemisApp.courseOverview.menu.messages' | artemisTranslate }}</span>
         </a>
         <a
-            *ngIf="course.isAtLeastTutor"
+            *ngIf="course.timeZone || course.isAtLeastInstructor"
```
Comment (JohannesStoehr) [inline src/main/webapp/app/course/manage/overview/course-management-card.component.html:257]: We currently don't load the tutorialGroupsConfiguration here and to keep this PR small I decided to only use the timeZone here, which is also a prerequisite for tutorials

**Review by konrad2002 — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]
Code

**Review by Strohgelaender — APPROVED** [review]

**Review by sarpsahinalp — APPROVED** [review]
Tested on ts3 works as expected.
Tutor side:
image

Instructor side:
Screenshot from 2023-10-24 19-02-33

**Pull request merged**